exampleAppKotlin/src/main/kotlin/it/trade/android/exampleapp/MainActivity.kt

```diff
@@ -506,6 +489,16 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    private fun getFirstCryptoAccount(
+        linkedBrokers: List<TradeItLinkedBrokerParcelable>
+    ): TradeItLinkedBrokerAccountParcelable? {
+        return linkedBrokers.mapNotNull { linkedBroker ->
```

Comment:
This can be simplified further so that it doesn't continue iterating once you have found the first crypto account:
```kotlin
val cryptoAccount = linkedBrokers.firstOrNull { linkedBroker ->
    linkedBroker.accounts.firstOrNull { account ->
        it.getOrderCapabilityForInstrument(Instrument.CRYPTO) != null
    } != null
}
```

(You can use `firstOrNull` or `find` as they both do the same thing if you pass a predicate)

Reply:
This doesn't compile. We need to return a `TradeItLinkedBrokerAccountParcelable`

Reply:
you'll need to use `map` or `mapNonNull` instead of `find` or `first`

Reply:
Ah sorry it should be:
```kotlin
val cryptoAccount = linkedBrokers.firstOrNull { linkedBroker ->
    linkedBroker.accounts.firstOrNull { account ->
        it.getOrderCapabilityForInstrument(Instrument.CRYPTO) != null
    } != null
}?.firstOrNull()
```